pkg/sidetree/cas/client.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cas
+
+import (
+	"bytes"
+	"crypto"
+	"encoding/base64"
+
+	"github.com/hyperledger/fabric/core/chaincode/shim"
+	"github.com/pkg/errors"
+)
+
+// New returns a new client for managing content
+func New(stub shim.ChaincodeStubInterface, collection string) *Client {
+	client := &Client{stub: stub, collection: collection}
+	client.Init(stub)
+	return client
+}
+
+// Init initializes the client
+func (mc *Client) Init(stub shim.ChaincodeStubInterface) {
+	mc.stub = stub
+}
+
+// Client implements writing and reading content
+type Client struct {
+	stub       shim.ChaincodeStubInterface
+	collection string
+}
+
+// Write stores content to DCAS.
+// returns the SHA256 hash in base64url encoding which represents the address of the content
+func (mc *Client) Write(content []byte) (string, error) {
+
+	address := calculateAddress(content)
+
+	payload, err := mc.stub.GetPrivateData(mc.collection, address)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to read content")
+	}
+	if payload != nil {
+		return "", errors.New("content already exists")
+	}
+
+	if err := mc.stub.PutPrivateData(mc.collection, address, content); err != nil {
```

Comment:
You can simply pass in an empty string for the key since it's always calculated as the hash of the value.

Reply:
I need to return address to the client.